source/hp/dof_handler.cc

```diff
@@ -227,7 +227,7 @@ namespace internal
           vertex_fe_association (dof_handler.finite_elements->size(),
                                  std::vector<bool> (dof_handler.tria->n_vertices(), false));
 
-          for (typename DoFHandler<dim,spacedim>::active_cell_iterator
+          for (TriaActiveIterator < dealii::DoFCellAccessor < DoFHandler < dim, spacedim >, false > >
```

Comment:
I understand. As I wrote in the pull request description, the library had already built just fine after the previous batch of changes. But we ran into runtime exceptions ExcCantCompareIterators etc. due to the same issue as before, that:

typename DoFHandler<dim,spacedim>::active_cell_iterator

in the hp namespace resulted in

TriaActiveIterator < dealii::DoFCellAccessor < dealii::DoFHandler < dim, spacedim >, false > >

(not in the hp namespace)

when we left the template argument decision to VS. If I changed it like I did in the code, it found the correct template argument and the error did not occur.

I know there has been a lot of work putting the template structure together and I would be happy if there were a nicer way around this problem to make it work with Visual Studio. Could you suggest something maybe?